etc/kayobe/environments/ci-aio/automated-setup.sh

```diff
@@ -1,109 +1,333 @@
-
-kayobe control host bootstrap
-
-kayobe playbook run etc/kayobe/ansible/growroot.yml etc/kayobe/ansible/purge-command-not-found.yml
-
-kayobe overcloud host configure
-
-kayobe overcloud service deploy
-
-if $AIO_RUN_TEMPEST; then
-    pushd $BASE_PATH/src/kayobe-config
+}
+
+# Ensure vault password file exists
+check_vault_password() {
+    if [[ ! -f "$VAULT_PASSWORD_FILE" ]]; then
+        error_exit "Vault password file not found at $VAULT_PASSWORD_FILE"
+    fi
+}
+
+# Ensure LVs are a minimum size. They will be extended during
+# host-configure, but need enough space to install kayobe first
+configure_lvm() {
+    if sudo vgdisplay | grep -q lvm2; then
+        echo "Resizing LVM volumes..."
+        sudo pvresize "$(sudo pvs --noheadings | head -n 1 | awk '{print $1}')"
+        sudo lvextend -L 4G /dev/rootvg/lv_home -r || true
+        sudo lvextend -L 4G /dev/rootvg/lv_tmp -r || true
+    elif [[ "$KAYOBE_AIO_LVM" == true ]]; then
+        error_exit "This environment is designed for images using the default StackHPC LVM layout. Set KAYOBE_AIO_LVM to false to ignore this warning."
+    fi
+}
+
+# Install package dependencies
+install_dependencies() {
+    echo "Installing dependencies..."
+    if type dnf > /dev/null 2>&1; then
+        sudo dnf -y install git python3.12
+    else
+        sudo apt update
+        sudo apt -y install gcc git libffi-dev python3.12-dev python-is-python3 python3.12-venv
+    fi
+}
+
+# Clone source repositories
+clone_repositories() {
+    echo "Cloning repositories..."
+    mkdir -p "$BASE_PATH/src"
+    pushd "$BASE_PATH/src"
+    # NOTE: Kayobe isn't used for deploying the AIO, but it is used for
+    # running the overcloud test VM script. TODO: run installed version
+    [[ -d kayobe ]] || git clone https://github.com/stackhpc/kayobe.git -b "$KAYOBE_BRANCH"
+    [[ -d kayobe-config ]] || git clone https://github.com/stackhpc/stackhpc-kayobe-config kayobe-config -b "$KAYOBE_CONFIG_BRANCH"
+    popd
+}
+
+# Pause for configuration editing if requested
+pause_for_configuration() {
+    if [[ "$KAYOBE_CONFIG_EDIT_PAUSE" == true ]]; then
+        echo "Deployment is paused, edit configuration in another terminal"
+        echo "Press enter to continue"
+        read -rs
+    fi
+}
+
+# Disable OVN if requested (aka deploy OVS)
+disable_ovn() {
+    if [[ "$USE_OVS" == true ]]; then
+        echo "Disabling OVN..."
+        echo "kolla_enable_ovn: false" > "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/disable-ovn.yml"
+    fi
+}
+
+# Remove LVM configuration if not using LVM
+remove_lvm_configuration() {
+    if ! sudo vgdisplay | grep -q lvm2; then
+        echo "Removing LVM configuration..."
+        rm -f "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/inventory/group_vars/controllers/lvm.yml" || true
+        sed -i -e '/controller_lvm_groups/,+2d' "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/controllers.yml" || true
+    fi
+}
```

Comment:
I think there should also be a check for `KAYOBE_AIO_LVM` here. Otherwise, you could run the script with `KAYOBE_AIO_LVM=true` and still have the LVM config removed. 

```suggestion
remove_lvm_configuration() {
    if ! sudo vgdisplay | grep -q lvm2; then
        echo "Removing LVM configuration..."
        rm -f "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/inventory/group_vars/controllers/lvm.yml" || true
        sed -i -e '/controller_lvm_groups/,+2d' "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/controllers.yml" || true
    elif [[ "$KAYOBE_AIO_LVM" == true ]]; then
        error_exit "You have KAYOBE_AIO_LVM set to true, but LVM is has not been found on this system. Set KAYOBE_AIO_LVM to false to ignore this warning."
cd $BASE_PATH
    fi
}
```

Reply:
I agree with the sentiment, though I think the check is in the wrong place in the example.
I'll push a new version which includes a slightly tweaked check

Reply:
Done